daemon/k8s_watcher.go

```diff
@@ -2174,3 +2172,23 @@ func (d *Daemon) deleteK8sNodeV1(k8sNode *v1.Node) {
 		return
 	}
 }
+
+// missingK8sNodeV1 checks if all nodes from the possible missing nodes have
+// their CiliumHostIP missing from the ipcache.
+func (d *Daemon) missingK8sNodeV1(m versioned.Map) versioned.Map {
+	missing := versioned.NewMap()
+	for k, v := range m {
+		n := v.Data.(*v1.Node)
+		ciliumHostIP := n.GetAnnotations()[annotation.CiliumHostIP]
```

Comment:
Can we just use `node.GetNode()`?

Reply:
@tgraf I'm a little lost here, how would it help?

Reply:
Well, it is a lookup function into the list of nodes that Cilium is aware of. the current implementation relies on the annotation of the node resource.

Reply:
The ipcache is also populated via the kvstore. Maybe I misunderstand the intention of the code here.

Reply:
The intention is simple, either from the kv-store or kubernetes notification events, we only want the ipcache to contain `ciliumHostIP`. That's what we are checking here.

Reply:
OK. Please explain to me why we add all of the complexity to lookup via `ciliumHostIP` if we have a perfect API to lookup via the provided hostname. This implementation is vulnerable to potentially miss identifying if the nodename <-> ciliumHostIP changes as the lookup via ciliumHostIP would still succeed whereas a proper node name lookup followed by key attribute comparison would catch all of that.